libs/ui-lib/lib/common/components/ui/formik/validationSchemas.ts

```diff
@@ -38,8 +38,8 @@ const CLUSTER_NAME_VALID_CHARS_REGEX = /^[a-z0-9-]*$/;
 const SSH_PUBLIC_KEY_REGEX =
   /^(ssh-rsa|ssh-ed25519|ecdsa-[-a-z0-9]*) AAAA[0-9A-Za-z+/]+[=]{0,3}( .+)?$/;
 const DNS_NAME_REGEX = /^([a-z0-9]+(-[a-z0-9]+)*\.)+[a-z]{2,}$/;
-const BASE_DOMAIN_REGEX = /^([a-z0-9]+(-[a-z0-9]+)*)$/;
-const DNS_NAME_REGEX_OCM = /^([a-z0-9]+(-[a-z0-9]+)*[.])+[a-z]{2,}$/;
+const BASE_DOMAIN_REGEX = /^[a-z\d][\-]*[a-z\d]+$/;
+const DNS_NAME_REGEX_OCM = /^([a-z\d]([\-]*[a-z\d]+)*\.)+[a-z\d]+[\-]*[a-z\d]+$/;
```

Comment:
This one is vulnerable to a REDoS attack. Here is the attack string: `'00' + '0'.repeat(31) + '00'`

Reply:
@jkilzi the regexp comes from BE. I've copied from https://github.com/openshift/assisted-service/pull/5371/files

Reply:
@gamli75 PTAL to this

Reply:
> This one is vulnerable to a REDoS attack. Here is the attack string: `'00' + '0'.repeat(31) + '00'`

@omertuc not sure if it's what you suggested

Reply:
Doesn't look like @jkilzi 's provided pattern is causing the backend to hang or take too long to process the request, so perhaps the Golang library for handling regex knows how to avoid the internal regex algorithms that lead to ReDOS.

I'm not familiar enough with ReDoS to confidently say we're in the clear though.

As for the UI, I don't think we should really care about REDoS, if a user wants to freeze up their own browser they can go ahead and do that

Reply:
> As for the UI, I don't think we should really care about ReDoS, if a user wants to freeze up their own browser they can go ahead and do that

@omertuc  This is not something the users do intentionally. Usually, it's by accident.

So if we can find a regex that avoids the issue and doesn't freeze the browser, we should.

Reply:
Yeah alright, fair point. Trying it out a little I saw that pretty much any long string without `.`'s in it would cause the regex to hang, and not just strings crafted in a particular way, so that's not ideal

Reply:
Or maybe the UI can use a safer regex library more similar to golang's: https://www.npmjs.com/package/re2

The package description mentions node but iiuc it should also work in browsers?

Reply:
**11.6 MB uncompressed**, that's the size of the node-re2 library. 
TL;DR: it is not suitable for web applications. I can elaborate more on why it is not a good fit but that will require a longer post.

Reply:
yikes